src/shogun/regression/LeastAngleRegressionImpl.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) The Shogun Machine Learning Toolbox
+ * Written (w) 2016 Saurabh Mahindre
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/Math.h>
+#include <shogun/regression/LeastAngleRegressionImpl.h>
+#include <shogun/mathematics/eigen3.h>
+
+using namespace std;
+using namespace shogun;
+using namespace Eigen;
+
+namespace shogun
+{
+SGMatrix<float64_t> CLeastAngleRegressionImpl::cholesky_insert_helper(SGMatrix<float64_t> X, SGMatrix<float64_t> R, VectorXd x_vec, SGVector<float64_t> active_set)
+{
+	if (!active_set.vector)
+	{
+		active_set=SGVector<float64_t>(X.num_cols);
+		active_set.range_fill();
+	}
+	int32_t num_active=active_set.size();
+	float64_t diag_k = x_vec.dot(x_vec);
+
+	// col_k is the k-th column of (X'X)	
+	SGVector<float64_t> col_k(num_active);
+	for (int32_t i=0; i < num_active; ++i)
+	{
+		// col_k[i] = X[:,i_max_corr]' * X[:,m_active_set[i]]
+		Map<VectorXd> active_i(X.get_column_vector(active_set[i]), X.num_rows);
+		col_k[i]=x_vec.dot(active_i);
+	}
+
+	// R' * R_k = (X' * X)_k = col_k, solving to get R_k
+	Map<VectorXd> col_vec(col_k.vector, col_k.size());
```

Comment:
These are updates of Cholesky factors. There are eigen3 methods for this...

How they are embedded within pseudo code, an example is here:
https://github.com/jbowlan/pyl1ls/blob/master/lars.py

Eigen3 methods are here:
https://eigen.tuxfamily.org/dox/classEigen_1_1LLT.html#a67b49e986da81c3c90fe1793681e70ad

It would be good if those would be put into linalg (later). But definitely not do this manually ....